File: KevinBankersen/ql/src/main/java/eu/bankersen/kevin/ql/ast/expr/logic/GT.java
package eu.bankersen.kevin.ql.ast.expr.logic;

import eu.bankersen.kevin.ql.ast.expr.Expr;

import com.esotericsoftware.minlog.Log;

import eu.bankersen.kevin.ql.ast.expr.BooleanExpr;

public class GT extends BooleanExpr {

    public GT(final Expr lhs, final Expr rhs) {
	super.lhs = lhs;
	super.rhs = rhs;
    }

    @Override
    public final Boolean eval() {
	Log.error("Made it here");
	
	return (Integer) lhs.eval() > (Integer) rhs.eval();
    }
}

File: Edward Poot/src/ql/ast/expr/binaryoperatorexpr/gt.go
package binaryoperatorexpr

import (
	"ql/ast/expr"
	"ql/ast/visit"
)

type GT struct {
	Lhs, Rhs expr.Expr
}

func (g GT) GetLhs() expr.Expr {
	return g.Lhs
}

func (g GT) GetRhs() expr.Expr {
	return g.Rhs
}

func (g GT) Eval() interface{} {
	return g.GetLhs().Eval().(int) > g.GetRhs().Eval().(int)
}

func (g GT) Accept(v visit.Visitor, s interface{}) interface{} {
	return v.Visit(g, s)
}